### ISSUE TITLE
challenge-sum-of-all-even-numbers

### ISSUE THREAD

**Issue by jiaosenvip** [labex-labs/scenarios#384]
> **Please read [For Newbie](https://github.com/labex-labs/scenarios/issues/18) first. 请先阅读 [新手须知](https://github.com/labex-labs/scenarios/issues/18)。**

## Basic

> Basic information about the scenario, which can be changed if required.
> 实验/挑战的基本信息，如有需要，可以修改。

- **Suggest Title 建议标题**: Sum of all Even Numbers
- **Suggest Steps 建议步骤数**: 1
- **Suggest Image 建议环境**: `webide-ubuntu:2204`

## Sources

> Issues has provided the material; the part not provided must be completed by the author.
> Issues 已经提供了材料；没有提供的部分由作者自行补充完成。

- [x] Full Text 正文内容
- [x] Reference Answers 参考答案
- [x] Verify Script 检测脚本

## Skills

> Suggested skills to include can be modified as needed. **The main skill points that should be included in this section are bolded**. From [Skills Tree](https://labex-dev.feishu.cn/base/bascnNz4Nqjqgqm1Nm5AYke6xxb?table=tblV5pGIsGZMxmE9&view=vewZpkLWRp).
> 建议包括的 Skills ID，可以根据需要增删。**本章节应包含的主要技能点已加粗**，技能点来自 [技能树](https://labex-dev.feishu.cn/base/bascnNz4Nqjqgqm1Nm5AYke6xxb?table=tblV5pGIsGZMxmE9&view=vewZpkLWRp)

- **`python/list_comprehension`**
- **`python/function_basic`**

**Comment by jiaosenvip** [comment]
# Sum of all Even Numbers

## Introduction

This challenge will test your knowledge in Python Functions. You will be required to implement a function that will perform a specific operation based on the requirements.

## Problem Description

You are given a list of integers. Your task is to write a Python function that will find the sum of all the even numbers in the list.

## Requirements

- Write a Python function named `even_sum` that takes a list of integers as an argument and returns the sum of the even numbers in the list.
- The function should ignore odd numbers and only consider even numbers.
- If the list is empty, the function should return 0.

## Example Usage

```python
even_sum([1, 2, 3, 4, 5, 6])
# Output: 12

even_sum([10, 15, 20])
# Output: 30

even_sum([3, 5, 7])
# Output: 0

even_sum([])
# Output: 0
```

## Code Template

```python
def even_sum(lst):
    pass
```

## Summary

In summary, you need to write a Python function named `even_sum` that takes a list of integers, finds the sum of all the even numbers in the list, and returns the result.

## Answer

```python
def even_sum(lst):
    return sum([i for i in lst if i % 2 == 0])
```

## Unittest

```python
import unittest

class TestEvenSum(unittest.TestCase):
    def test_even_sum(self):
        self.assertEqual(even_sum([1, 2, 3, 4, 5, 6]), 12)
        self.assertEqual(even_sum([10, 15, 20]), 30)
        self.assertEqual(even_sum([3, 5, 7]), 0)
        self.assertEqual(even_sum([]), 0)

if __name__ == '__main__':
    unittest.main()
```
